### PR TITLE
Gather more summary data when running `result_analyzer.py`.

### PR DESCRIPTION
**Pull request by ysiraichi**
This PR refactors and computes more summary data out of each timing sequence. More specifically, besides the median, we also compute and store: min, max, mean, and std.

### CODE REVIEW

**Comment by ysiraichi** [comment]
cc @JackCaoG @miladm

**Review by frgossen — CHANGES_REQUESTED** [review]
Thanks for adding these. I realized that the `result_analyzer` is broken for some configurations, e.g. producing csv output. Would you mind adding a test for your change? It'd be great to have a start so we can all add tests over time.

**Comment by vanbasten23** [comment]
Wondering if you have compared the results before and after the change.

**Review by vanbasten23 — APPROVED** [review]

**Comment by ysiraichi** [comment]
Here's a screenshot of the difference of one benchmark run. Note that, even though it's still there, `xla_median_trace_per_iter_time` was replaced by `median_trace_per_iter_time`.

diff

**Comment by ysiraichi** [comment]
I propose that we refactor this script a bit further, so that:

- it summarizes the data from JSONL, into the same fieldnames
    - do we actually need the JSON output version?
- we don't need to explicitly describe the header

**Comment by frgossen** [comment]
> I propose that we refactor this script a bit further, so that:
> 
> * it summarizes the data from JSONL, into the same fieldnames
With "the same fieldnames", do you mean to write it back to that original file? I think having this analysis as a second step is a good thing. We should eventually merge this with the aggregate script though 

>   * do we actually need the JSON output version?
Unless we have uses for both, it'd be good to have either CSV or JSON, not both. I have no preference. The aggregate script is based on CSV atm. 

> * we don't need to explicitly describe the header
I assume this is for CSV files? It's probably a good idea to have a header line to keep it readable.

**Comment by ysiraichi** [comment]
> With "the same fieldnames", do you mean to write it back to that original file?

No, I mean the headers. At the moment, CSV and JSON outputs have different key-value pairs (JSON is hierarchically laid out).

> I assume this is for CSV files? It's probably a good idea to have a header line to keep it readable.

Yes. I mean, in the script, [we explicitly write the headers using `pd.Dataframe`](https://github.com/pytorch/xla/blob/fe4af0080af07f78ca2b614dd91b71885a3bbbb8/benchmarks/result_analyzer.py#L49). Instead, we can make the script work with whatever data was collected, without explicitly describing them in the **script**. The output would still have a headerline.

**Review by golechwierowicz — CHANGES_REQUESTED** [review]

File: benchmarks/result_analyzer.py [hunk 1]
```diff
@@ -74,41 +74,32 @@ def run_csv(self):
...
-      d["median_total_cpu_time_s"] = np.median(total_cpu_time)
-    if "per_iter_cpu_time_s" in dataline["metrics"]:
-      per_iter_cpu_time = np.asarray(
-          dataline["metrics"]["per_iter_cpu_time_s"], dtype="float")
-      d["median_per_iter_cpu_time_s"] = np.median(per_iter_cpu_time)
-    if "total_cuda_time_s" in dataline["metrics"]:
-      total_cuda_time = np.asarray(
-          dataline["metrics"]["total_cuda_time_s"], dtype="float")
-      d["median_total_cuda_time_s"] = np.median(total_cuda_time)
-    if "per_iter_cuda_time_s" in dataline["metrics"]:
```
[the remark this answers is not in the document]
Comment (golechwierowicz) [reply benchmarks/result_analyzer.py:103]: You're right. I was surprised not to see the crude code :P Can you add a test to cover whether device time is analyzed?

**Review by golechwierowicz — APPROVED** [review]

**Pull request closed without merging**